crypto/keys/src/x25519.rs

```diff
@@ -259,7 +279,7 @@ impl<'de> Deserialize<'de> for X25519Public {
             }
 
             fn visit_bytes<E: DeserializeError>(self, value: &[u8]) -> Result<Self::Value, E> {
-                X25519Public::try_from_der(value).map_err(|err| E::custom(err.to_string()))
+                X25519Public::try_from_der(value).map_err(|err| E::custom(err))
```

Comment:
Equivalently, pass the method directly
```suggestion
                X25519Public::try_from_der(value).map_err(E::custom)
```